bfd/elflink.c

```diff
@@ -6580,7 +6580,6 @@ elf_gc_sweep_symbol (struct elf_link_hash_entry *h, void *data)
 
       inf = (struct elf_gc_sweep_symbol_info *) data;
       (*inf->hide_symbol) (inf->info, h, true);
-      h->def_regular = 0;
```

Comment:
doesn't look microblaze related; to be inspected

Reply:
upstream change to garbage collection sweep causes mb regression

Reply:
Sounds like it fixes an issue. Keeping it in.

Reply:
> doesn't look microblaze related; to be inspected

That was exactly my concern -- I was going to leave a comment about it then got interrupted by something else and it went off my radar ... (sorry about that).

My concern is its potential effect on other architectures -- will it not cause any regressions on other architectures relying on `def_regular` to be set to `0` by this function?

Unless we can prove it will not cause any regressions for other architectures, could we make this specific to Microblaze only? (maybe make the function accept `bfd` param and add this under `if (bfd_get_arch(abfd) == bfd_arch_microblaze)`?)

Reply:
I think I'll undo the patch for the time being because also from the original commit message, this seems very non-zephyr+embedded-linux related. If this gets upstreamed,                   and then if it finds its way down here through upstream `binutils`, then sure. 
>The visible issue with this change is when running a c++ application
in Petalinux which links libstdc++.so for exception handling it segfaults
on execution.
This does not occur if static linking libstdc++.a, so its during the
relocations for a shared lib with garbage collection this occurs
I'll now take a look at the others too.